src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/TrackerObjectManager.NativeAot.cs

```diff
@@ -0,0 +1,543 @@
+            vftbl[0] = (IntPtr)(delegate* unmanaged<IntPtr, Guid*, IntPtr*, int>)&IFindReferenceTargetsCallback_QueryInterface;
+            vftbl[1] = (IntPtr)(delegate* unmanaged<IntPtr, uint>)&ComWrappers.Untracked_AddRef;
+            vftbl[2] = (IntPtr)(delegate* unmanaged<IntPtr, uint>)&ComWrappers.Untracked_Release;
+            vftbl[3] = (IntPtr)(delegate* unmanaged<IntPtr, IntPtr, int>)&IFindReferenceTargetsCallback_FoundTrackerTarget;
+            return (IntPtr)vftbl;
+        }
+
+        internal static unsafe IntPtr CreateFindReferenceTargetsCallback()
+        {
+            IntPtr wrapperMem = (IntPtr)NativeMemory.Alloc(2 * (nuint)sizeof(IntPtr));
+            *(IntPtr*)(wrapperMem) = CreateDefaultIFindReferenceTargetsCallbackVftbl();
+            *(IntPtr*)(wrapperMem + sizeof(IntPtr)) = wrapperMem;
+            return wrapperMem;
+        }
+    }
+
+    internal readonly struct ComHolder : IDisposable
+    {
+        private readonly IntPtr _ptr;
+
+        internal readonly IntPtr Ptr => _ptr;
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public ComHolder(IntPtr ptr)
+        {
+            _ptr = ptr;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public readonly void Dispose()
+        {
+            Marshal.Release(_ptr);
+        }
+    }
+
+    // This is used during a GC callback so it needs to be free of any managed allocations.
+    internal unsafe struct DependentHandleList
+    {
+        private int _freeIndex;                // The next available slot
+        private int _capacity;                 // Total numbers of slots available in the list
+        private IntPtr* _pHandles;             // All handles
+        private int _shrinkHint;               // How many times we've consistently seen "hints" that a
+                                               // shrink is needed
+
+        private const int DefaultCapacity = 100;       // Default initial capacity of this list
+        private const int ShrinkHintThreshold = 10;    // The number of hints we've seen before we really
+                                                       // shrink the list
+
+        public bool AddDependentHandle(object target, object dependent)
+        {
+            if (_freeIndex >= _capacity)
+            {
+                // We need a bigger dependent handle array
+                if (!Grow())
+                    return false;
+            }
+
+            IntPtr handle = _pHandles[_freeIndex];
+            if (handle != default)
+            {
+                RuntimeImports.RhHandleSet(handle, target);
+                RuntimeImports.RhHandleSetDependentSecondary(handle, dependent);
+            }
+            else
+            {
+                _pHandles[_freeIndex] = RuntimeImports.RhHandleAllocDependent(target, dependent);
```

Comment:
Do we need to check that this succeded?

Reply:
It looks like it will throw if it fails.  I think for an out of memory allocating a handle that might be fine to leave as is and break there instead of catching or adding another import that doesn't throw.  Let me know if there are other concerns here where we should indeed catch the exception.

Reply:
> It looks like it will throw if it fails

It is not going to run that far. This code runs during GC where allocations are disallowed. It will lead to some very bad crash.

This should call the lower-level non-throwing method.

Reply:
Addressed this and a couple other exceptions on other call paths from GC callbacks.